docs/signing/local-signing.md

```diff
@@ -0,0 +1,151 @@
+
+The rest of this tutorial uses OpenSSL (a set of cryptographic utilities). If OpenSSL is not installed on your system, see [OpenSSL](https://www.openssl.org/source/) for the source distribution or the [list of unofficial binary distributions](https://wiki.openssl.org/index.php/Binaries).
+
+### Step 2: Extract the certificate and key
+
+To work with the certificate, you need to extract it. When the CAI SDK adds Content Credentials to an asset, it incorporates the certificate (including the associated public key) into the manifest.
+Use the commands below to extract the key and certificate chain. If prompted, enter the password that was used to generate the `.pfx` file.
+
+:::tip
+Make sure you are using a recent version of OpenSSL.
+:::tip
+
+#### Troubleshooting errors
+
+In this step, OpenSSL may report errors when extracting the key or certificate chain.  In many cases, if OpenSSL generates the output file, you can ignore the messages.  However, in some cases you may need to add `-legacy` to the command for it to work properly.
+
+For example, the following error message means the `.pfx` was encrypted with an older standard:
+
+```
+Shrouded Keybag: pbeWithSHA1And3- KeyTripleDES-CBC, Iteration 2000
+PKCS7 Encrypted data: pbeWithSHA1And40BitRC2- CBC, Iteration 2000
+Error outputting keys and certificates
+```
+
+#### Extract the key
+
+```shell
+openssl pkcs12 -in mycertfile.pfx -nocerts -out mykey.pem -nodes
+```
+
+:::tip
+Check to make sure the above command generated a `.pem` file and it's not an empty file.  For more information, see [Troubleshooting errors](#troubleshooting-errors) above.
+:::
+
+#### Extract the certificate chain
+
+For many certificate providers, the `.pfx` file contains not just your certificate but the complete certificate trust chain. When the `.pfx` file does not contain the certificate chain, you can obtain it from your provider.
+
+```shell
+openssl pkcs12 -in mycertfile.pfx -nokeys -out mycerts.pem
+```
+
+### Step 3: Use credentials with C2PA Tool
+
+To use the credentials extracted above you must know the signature types they support. Typically, the certificate provider will provide this information. If it is not, enter this OpenSSL command to dump certificate information:
+
+```shell
+openssl x509 -inform PEM -in mycerts.pem -text
+```
+
+This command produces a text summary of the certificate properties, as shown in the example below. Look for a line containing `Signature Algorithm`. See the table above to determine the corresponding signature type. For this example with a certificate issued by GlobalSign, `Signature Algorithm: sha256WithRSAEncryption` corresponds to the PS256 signature type.
```

Comment:
This section is not correct.  If you dump the cert  as above you should be looking in the 
**Subject Public Key Info:
Public Key Algorithm:**

The public key determines what types of signatures you can support:

So for PS signatures you will see rsassaPss public key type and the parameters to determining the hash for example
 PSS parameter restrictions:
                  Hash Algorithm: SHA2-256
                  Mask Algorithm: MGF1 with SHA2-256
                  Minimum Salt Length: 32
                  Trailer Field: 0x1 (default)

This would be PS256

For EC certs you will see: id-ecPublicKey for public key type with curve value like P-256, e.g.
This would correspond to ES256

For ED25519 you will see: ED25519 for the public key type which maps to ED25519

Reply:
Jeez, this has been in the docs for 2.5 years!  It came from a PR from Kozma that you and @gpeacock  reviewed: https://github.com/contentauth/opensource.contentauth.org/pull/52 


Well, water under the bridge. I'll work on fixing it now.

Reply:
Edited based on above comments.

Reply:
Made additional changes using input from Slack convo.